### PR TITLE
[ticket/15889] Add core.memberlist_modify_memberrow_sql

### PR DESCRIPTION
**Pull request by 3D-I**
PHPBB3-15889

Checklist:

- [x] Correct branch: master for new features; 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.2.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-15889

### CODE REVIEW

**Comment by 3D-I** [comment]
!set Event

**Comment by 3D-I** [comment]
!set WIP :construction:

**Comment by 3D-I** [comment]
Sorry for the lot of commits. Now I think I am done here.

File: phpBB/memberlist.php [hunk 1]
```diff
@@ -1469,19 +1469,46 @@
...
+			/**
+			 * Modify user data SQL before member row is created
+			 *
+			 * @event core.memberlist_modify_memberrow_sql
+			 * @var string	mode				group
+			 * @var string	sql_select			Additional select statement
+			 * @var string	sql_from			Additional from statement
+			 * @var array	sql_array			Array containing the main query
+			 * @var array	user_list			Array containing list of users
+			 * @since 3.2.6-RC1
```
Comment (dark-1) [inline phpBB/memberlist.php]: Targeting `3.2.6` OR `3.2.5` ?

File: phpBB/memberlist.php [hunk 2]
```diff
@@ -1492,9 +1519,9 @@
 
 				$id_cache[$row['user_id']] = $row;
 			}
+
 			$db->sql_freeresult($result);
 
-			// Load custom profile fields
```
Comment (dark-1) [inline phpBB/memberlist.php]: Why remove this Comment ? 🤔

Comment (3D-I) [reply phpBB/memberlist.php]: Good catch.

**Comment by marc1706** [comment]
@3D-I Added a minor modification to your PR. I think this should result in a proper format for the `'FROM'` entry in the sql array.

**Comment by 3D-I** [comment]
@marc1706 :+1: 

Something went wrong with Travis though

https://travis-ci.org/phpbb/phpbb/jobs/473281294#L540

**Comment by 3D-I** [comment]
@marc1706 

Updated to 3.2.5.

**Comment by 3D-I** [comment]
This useless error has become recurrent/redundant, closing/opening the PR(s) so to let the tests start again usually does the trick, FYI. Quite boring though :smile_cat: 

https://travis-ci.org/phpbb/phpbb/jobs/475883395#L1223

File: phpBB/memberlist.php [hunk 3]
```diff
@@ -1469,19 +1469,58 @@
...
+					),
+					'WHERE'		=> $db->sql_in_set('u.user_id', $user_list),
+				);
 			}
+
+			/**
+			 * Modify user data SQL before member row is created
+			 *
+			 * @event core.memberlist_modify_memberrow_sql
+			 * @var string	mode				group
```
Comment (marc1706) [inline phpBB/memberlist.php]: It's not really the group, is it?

Comment (3D-I) [reply phpBB/memberlist.php]: I pushed the change, dunno why is not listed here below?

**Comment by 3D-I** [comment]
Hopefully will be squashed somehow in just one commit, tried to no avail.

**Pull request merged**